### PR TITLE
Upgrade to MUIv5

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
I followed the guide here: https://mui.com/material-ui/migration/migration-v4/
This will unblock us from upgrading to React 18

Mostly to resolve tech debt.

There are some visual changes:
- Links are underlined by default
- Input boxes are "outlined" variant by default

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
Any updates on this one?

**Review by GeneDer — APPROVED** [review]
LGTM!

File: dashboard/client/src/components/ActorTable.component.test.tsx [hunk 1]
```diff
@@ -344,57 +335,18 @@ describe("ActorTable", () => {
...
-      name: /ACTOR_1/,
-    });
-    const actor2GRAMRow = getByRole("row", {
-      name: /ACTOR_2/,
-    });
-    expect(within(actor1GRAMRow).getByText("ACTOR_1")).toBeInTheDocument();
-    expect(within(actor2GRAMRow).getByText("ACTOR_2")).toBeInTheDocument();
-    expect(actor2GRAMRow.compareDocumentPosition(actor1GRAMRow)).toBe(
-      Node.DOCUMENT_POSITION_FOLLOWING,
-    ); // actor2Row appear before actor1Row
```
Comment (architkulkarni) [inline dashboard/client/src/components/ActorTable.component.test.tsx:398]: Why delete the GPU / GRAM tests?

File: dashboard/client/src/components/ActorTable.tsx [hunk 2]
```diff
@@ -327,7 +327,7 @@ const ActorTable = ({
 
   return (
     <React.Fragment>
-      <div style={{ flex: 1, display: "flex", alignItems: "center" }}>
+      <div style={{ flex: 1, display: "flex", alignItems: "end" }}>
```
Comment (architkulkarni) [inline dashboard/client/src/components/ActorTable.tsx:330]: Is this required for the migration or just a style change? Are there other non-required changes in the PR? Would prefer to keep this PR just scoped to the necessary changes for the migration, to make it easier to review the style changes separately

File: dashboard/client/src/App.tsx
Comment (architkulkarni) [inline dashboard/client/src/App.tsx:1]: Can you summarize the changes here? It was a bit hard to read the diff

Comment (architkulkarni) [reply dashboard/client/src/App.tsx:1]: Never mind, this is easier to read https://app.semanticdiff.com/gh/ray-project/ray/pull/42454/files#dashboard/client/src/App.tsx

**Review by architkulkarni — APPROVED** [review]
Looks good, thanks for the effort. A few minor questions but they don't need to block the PR

File: dashboard/client/src/components/ActorTable.tsx (hunk 2 above)
Comment (alanwguo) [reply dashboard/client/src/components/ActorTable.tsx:330]: This is necessary to make things look good after the migration. There seems to be slight layout issues with the new UI components.

File: dashboard/client/src/components/ActorTable.component.test.tsx (hunk 1 above)
Comment (alanwguo) [reply dashboard/client/src/components/ActorTable.component.test.tsx:398]: This was because the tests were timing out before. Let me add them back.

File: dashboard/client/src/components/ActorTable.tsx (hunk 2 above)
Comment (architkulkarni) [reply dashboard/client/src/components/ActorTable.tsx:330]: Gotcha, thanks
